src/frontend/src/pages/PIADetailPage/index.tsx

```diff
@@ -71,13 +90,47 @@ const PIADetailPage = () => {
         }
       }
     })();
-  }, [id, navigate, pia.hasAddedPiToDataElements, pia.ministry, win]);
+  }, [
+    id,
+    navigate,
+    pia.hasAddedPiToDataElements,
+    pia.ministry,
+    pia.status,
+    win,
+  ]);
 
   const handleAlertClose = () => {
     setFetchPiaError('');
     setDownloadError('');
   };
 
+  const handleEdit = () => {
+    if (isMPO) {
+      setModalConfirmLabel(messages.Modal.ConfirmLabel.en);
+      setModalCancelLabel(messages.Modal.CancelLabel.en);
+      setModalTitleText(messages.Modal.TitleText.en);
+      setModalParagraph(messages.Modal.ParagraphText.en);
+      setShowModal(true);
+    } else {
+      navigate(routes.PIA_INTAKE, {
+        state: pia,
+      });
+    }
+  };
+
+  const handleModalClose = () => {
+    setShowModal(false);
+    navigate(routes.PIA_INTAKE, {
+      state: pia,
+    });
+  };
+  const handleModalCancel = () => {
+    setShowModal(false);
+  };
+  const handleSubmit = () => {
+    console.log('will do');
```

Comment:
this function will be refactor in the next ticket

Reply:
then I would recommend commenting this and the actual button code; That way we will prevent showing the button that does not work at all.